content/blog/esp32-memory-map-101/index.md

```diff
@@ -0,0 +1,351 @@
+
+
+Zephyr-RTOS support for the ESP32-S3 devkit board can be found [here](https://docs.zephyrproject.org/latest/boards/espressif/esp32s3_devkitc/doc/index.html).
+
+**ROM**
+
+ROM memories hold the primary bootloader code and other library functions available to the end users. The content of the ROM memories is baked during the manufacturing process and cannot be changed.
+
+**SRAM**
+
+There are three internal SRAM modules. The SRAM0 can be configured as an I-Cache buffer to access instructions from external memory or be used as static r/w memory via the instruction bus. SRAM1 is the r/w memory which can be accessed via the instruction and data bus. SRAM2 can be configured as D-Cache, or it can be accessed as static r/w memory via the data bus
+
+**RTC-SRAM**
+
+RTC-Slow is r/w memory that can be accessed via the instruction and data bus. RTC-Fast is the r/w memory that can be accessed via the instruction and data bus on a given address range.
+
+**Instruction & data cache**
+
+The CPUs can access external memories via the cached MMU address space. Memories can be accessed using I-Cache or D-Dache via instruction or data bus respectively.
+
+
+### Internal memories and cache address space
+
+{{< figure
+    default=true
+    src="img/esp32s3-mmap.webp"
+    alt=""
+    caption="The ESP32 memory map."
+    >}}
+
+
+### Peripherals
+
+The peripherals (modules) can be accessed at their address range using the data and instruction bus. Each peripheral module has its own memory boundaries given by its base address and block size. Blocks size is usually 4kB long. For more details refer to device TRM.
+
+{{< figure
+    default=true
+    src="img/esp32s3-periph.webp"
+    alt=""
+    caption="The ESP32-S3 peripherals."
+    >}}
+
+### eFuse blocks
+
+The eFuse memory consists of 4096 bits of OTP memory where hardware and encryption settings are stored. Fundamentally, an eFuse is a single bit of non-volatile memory with the restriction that once an eFuse bit is programmed to 1, it can never be reverted to 0.
+
+{{< figure
+    default=true
+    src="img/esp32s3-efuse.webp"
+    alt=""
+    caption="The ESP32-S3 e-fuse block."
+    >}}
+
+
+For further information and all the details about the ESP32-S3 SoC please refer to the latest [ESP32-S3 Technical Reference Manual](https://www.espressif.com/sites/default/files/documentation/esp32-s3_technical_reference_manual_en.pdf).
+
+---
+
+
+## ESP32-C3
+
+The ESP32-C3 is a 32-bit RISC-V microcontroller with the “IMC” extensions. The address space of ESP32-C3 is significantly simpler compared to its Xtensa counterparts. It supports JTAG debugging via the on-chip USB interface.
```

Comment:
The abbreviation IMC seems unfamiliar to me as well. There is likelihood that it can be new to readers.

Suggest that you expand it and, if it is not self-explanatory, maybe explain what it is.

Reply:
Since I wasn't able to find a source of documentation on the RISC-V extension that I could link as an explanation, I removed the reference altogether. It is not important in the context of the article.